miditoolkit/midi/parser.py

```diff
@@ -338,28 +349,37 @@ def dump(self,
 
         # comparison function
         def event_compare(event1, event2):
+            if event1.time != event2.time:
+                return event1.time - event2.time
+
+            # If its two note_on (at the same tick), sort by expected note_off in a FIFO logic
+            # This is required in case where the MIDI has notes starting at the same tick and one
+            # with a higher duration is listed before one with a shorter one. In this case, the note
+            # with the higher duration should come after, otherwise it will be ended first by the
+            # following note_off event. Ultimately, as the notes have the same starting time and pitch,
+            # the only thing that could be missed is their velocities. This check prevents this.
+            if event1.type == event2.type == "note_on":
+                return event1.end - event2.end
+
             secondary_sort = {
-                'set_tempo': lambda e: (1 * 256 * 256),
-                'time_signature': lambda e: (2 * 256 * 256),
-                'key_signature': lambda e: (3 * 256 * 256),
-                'marker': lambda e: (4 * 256 * 256),
-                'lyrics': lambda e: (5 * 256 * 256),
-                'program_change': lambda e: (6 * 256 * 256),
-                'pitchwheel': lambda e: ((7 * 256 * 256) + e.pitch),
-                'control_change': lambda e: (
-                        (8 * 256 * 256) + (e.control * 256) + e.value),
-                'note_off': lambda e: ((9 * 256 * 256) + (e.note * 256)),
-                'note_on': lambda e: (
-                        (10 * 256 * 256) + (e.note * 256) + e.velocity),
-                'end_of_track': lambda e: (11 * 256 * 256)
+                "set_tempo": 1,
```

Comment:
Sorting events of the same type at the same tick isn't useful as it doesn't alter the musical content.
Removing this allows to speed up the sorting by only using the message types.